server/aws-lsp-codewhisperer/src/language-server/amazonQServiceManager/configurationUtils.ts

```diff
@@ -0,0 +1,186 @@
+import { Logging, Lsp, Runtime } from '@aws/language-server-runtimes/server-interface'
+import { Q_CONFIGURATION_SECTION } from '../configuration/qConfigurationServer'
+import { textUtils } from '@aws/lsp-core'
+import {
+    AWS_Q_ENDPOINT_URL_ENV_VAR,
+    AWS_Q_ENDPOINTS,
+    AWS_Q_REGION_ENV_VAR,
+    DEFAULT_AWS_Q_ENDPOINT_URL,
+    DEFAULT_AWS_Q_REGION,
+} from '../../constants'
+
+export interface AWSQRegionAndEndpoint {
+    region: string
+    endpoint: string
+}
+
+/**
+ * Determines Q region according to the following discovery chain:
+ *
+ * provided region (e.g. by client or profile) -> region set in runtime -> default region
+ *
+ * @returns region and endpoint for Q connection
+ */
+export function getAWSQRegionAndEndpoint(runtime: Runtime, logging: Logging, region?: string): AWSQRegionAndEndpoint {
+    let awsQRegion: string
+    let awsQEndpoint: string | undefined
+
+    if (region) {
+        logging.log(`Selecting region (found: ${region}) provided by caller`)
+        awsQRegion = region
+        // @ts-ignore
+        awsQEndpoint = AWS_Q_ENDPOINTS[awsQRegion]
+    } else {
+        const runtimeRegion = runtime.getConfiguration(AWS_Q_REGION_ENV_VAR)
+
+        if (runtimeRegion) {
+            logging.log(`Selecting region (found: ${runtimeRegion}) provided by runtime`)
+            awsQRegion = runtimeRegion
+            // @ts-ignore
+            awsQEndpoint = runtime.getConfiguration(AWS_Q_ENDPOINT_URL_ENV_VAR) ?? AWS_Q_ENDPOINTS[awsQRegion]
+        } else {
+            logging.log('Region not provided by caller or runtime, falling back to default region and endpoint')
+            awsQRegion = DEFAULT_AWS_Q_REGION
+            awsQEndpoint = DEFAULT_AWS_Q_ENDPOINT_URL
+        }
+    }
+
+    if (!awsQEndpoint) {
+        logging.log(
+            `Unable to determine endpoint (found: ${awsQEndpoint}) for region: ${awsQRegion}, falling back to default region and endpoint`
+        )
+        awsQRegion = DEFAULT_AWS_Q_REGION
+        awsQEndpoint = DEFAULT_AWS_Q_ENDPOINT_URL
+    }
+
+    return {
+        region: awsQRegion,
+        endpoint: awsQEndpoint,
+    }
+}
+
+export interface QConfig {
+    customizationArn: string | undefined // aws.q.customizationArn - selected customization
+    optOutTelemetryPreference: 'OPTOUT' | 'OPTIN' // aws.q.optOutTelemetry - telemetry optout option
+    inlineSuggestions: {
+        extraContext: string | undefined // aws.q.inlineSuggestions.extraContext
+    }
+}
+
+export interface CodeWhispererConfig {
+    includeSuggestionsWithCodeReferences: boolean // aws.codeWhisperer.includeSuggestionsWithCodeReferences - return suggestions with code references
+    shareCodeWhispererContentWithAWS: boolean // aws.codeWhisperer.shareCodeWhispererContentWithAWS - share content with AWS
+}
+
+export interface AWSQWorkspaceConfigs {
```

Comment:
I think we can put everything under single Q config. `aws.codeWhisperer` is just a legacy name for `aws.q`.